src/pages/ToDoList/List.js

```diff
@@ -1,34 +1,78 @@
-import React from 'react';
+import React, {useState, useMemo} from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import ListGroup from '../../components/Group/Group';
 import AddGroupForm from '../../components/AddGroupForm'
 import styles from './List.module.css'
 import GroupsFilter from '../../components/GroupsFilter';
 import MyModal from '../../components/ui/modal/MyModal';
 import MyButton from '../../components/ui/button/MyButton';
+import { fetchGroups, fetchRemoveGroup } from '../../redux/slices/groups';
+import { selectIsAuth } from '../../redux/slices/auth';
+import { Link, Navigate } from 'react-router-dom';
 
-const List = (props) => {
+const List = ({setStatistics}) => {
+    const isAuth = useSelector(selectIsAuth);
+    const dispatch = useDispatch();
+    const { groups } = useSelector(state => state.groups);
+    const isGroupsLoading = groups.status === 'loading';
+    React.useEffect(() => {
```

Comment:
After importing `useEffect`:
```suggestion
    useEffect(() => {
```